### PR TITLE
Add an EditorConfig file

### PR DESCRIPTION
**Pull request by stmontgomery**
Add an EditorConfig (`.editorconfig`) file to this repo.

### Motivation:

[EditorConfig](https://editorconfig.org) helps maintain consistent coding styles for multiple developers working on the same project across various editors and IDEs. It will be helpful to add one of these files to swift-testing so that all contributors use the whitespace conventions described in our [style guide](https://github.com/swiftlang/swift-testing/blob/main/Documentation/StyleGuide.md) automatically when using a supported IDE/editor.

Note that Xcode 16 Beta has added support for EditorConfig files — see 20796230 in the [release notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-16-release-notes).

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

**Comment by stmontgomery** [comment]
@swift-ci please test

**Comment by stmontgomery** [comment]
@swift-ci please test Linux

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+root = true
+
+# Default settings applied to every file type.
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+insert_final_newline = true
+max_line_length = 80
```
Comment (grynspan) [inline .editorconfig:11]: We allow wider lines of code, we just wrap comments at 80 columns.

Comment (stmontgomery) [reply .editorconfig:11]: It seems `max_line_length` doesn't formally distinguish code vs comments, so I can leave this directive out of the file for now.

**Comment by stmontgomery** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
@swift-ci please test

**Review by grynspan — APPROVED** [review]
